src/ui/widgets/SlideControl/slideControl.tsx

```diff
@@ -1,115 +1,177 @@
-  }
+  useEffect(() => {
+    if (value) {
+      setInputValue(value?.getDoubleValue() ?? 0);
+    }
+  }, [value]);
 
-  function onMouseDown(event: React.MouseEvent<HTMLInputElement>): void {
-    setEditing(true);
-  }
-  function onMouseUp(event: React.MouseEvent<HTMLInputElement>): void {
-    setEditing(false);
+  function onMouseUp(value: number): void {
     try {
-      const doubleValue = parseFloat(event.currentTarget.value);
-      writePv(pvName, new DType({ doubleValue: doubleValue }));
+      writePv(pvName, new DType({ doubleValue: value }));
     } catch (error) {
-      log.warn(`Unexpected value ${event.currentTarget.value} set to slider.`);
+      log.warn(`Unexpected value ${value} set to slider.`);
     }
   }
 
-  const stringValue = DType.coerceString(value);
-  if (!editing && inputValue !== stringValue) {
-    setInputValue(stringValue);
-  }
+  const marks = [
+    ...(showHihi
+      ? [
+          {
+            value: levelHihi,
+            label: levelHihi.toString()
+          }
+        ]
+      : []),
+    ...(showHigh
+      ? [
+          {
+            value: levelHigh,
+            label: levelHigh.toString()
+          }
+        ]
+      : []),
+    ...(showLow
+      ? [
+          {
+            value: levelLow,
+            label: levelLow.toString()
+          }
+        ]
+      : []),
+    ...(showLolo
+      ? [
+          {
+            value: levelLolo,
+            label: levelLolo.toString()
+          }
+        ]
+      : [])
+  ];
 
   return (
-    <div>
-      <div
-        style={{
-          display: "block",
-          position: "relative",
-          height: "90%",
-          width: "100%",
-          top: "0%",
-          left: "0%"
-        }}
-      >
-        <ProgressBarComponent
-          connected={connected}
-          value={value}
-          min={min}
-          max={max}
-          limitsFromPv={limitsFromPv}
-          precision={precision}
-          readonly={props.readonly}
-          showLabel={true}
-        />
-      </div>
-      <div
-        style={{
-          display: "block",
-          position: "relative",
-          height: "10%",
-          width: "100%",
-          bottom: "0%",
-          left: "0%"
-        }}
-      >
-        <input
-          className={`Slider ${classes.Slider}`}
-          type="range"
-          min={min}
-          max={max}
-          value={inputValue}
-          onChange={onChange}
-          onMouseDown={onMouseDown}
-          onMouseUp={onMouseUp}
-        ></input>
-      </div>
-    </div>
+    <Slider
+      value={inputValue}
+      disabled={disabled}
+      orientation={horizontal ? "horizontal" : "vertical"}
+      onChange={(_, newValue) => setInputValue(newValue as number)}
+      onChangeCommitted={(_, newValue) => onMouseUp(newValue as number)}
+      valueLabelDisplay="auto"
+      min={min}
+      max={max}
+      marks={marks}
+      step={increment}
+      sx={{
+        color: backgroundColor.toString(),
```

Comment:
This widget seems a bit odd in Phoebus and sets the colour of the slider based off the `foregroundColor` rather than `backgroundColor`, which I think we should keep doing. Having the entire slider be a single colour looks better than just the thumb part.